asset-transfer-basic/chaincode-go/assetTransfer.go

```diff
@@ -58,7 +58,7 @@ func (s *SmartContract) InitLedger(ctx contractapi.TransactionContextInterface)
 }
 
 // CreateAsset issues a new asset to the world state with given details.
-func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface, id, color, owner string, size, appraisedValue int) error {
+func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface, id, color, size, owner string, appraisedValue int) error {
```

Comment:
I assumed the args in this chaincode-go is correct, and was trying to fix the arg sequence in chaincode-js & chaincode-typescript. In this PR #226 .

I assume we have only one application using these chaincodes: `asset-transfer-basic/application-javascript`

Reply:
You are correct, the args are correct but out of sequence vs the struct in those two functions.  The CreateAsset function and the UpdateAsset function had owner: string, and size: int reversed.  I only looked at the go sample as I was testing it in preparation to refactor the chaincode for developers tutorial.

Reply:
I changed the Typescript and Javascript applications to match.  If everything gets merged, it will be consistent across all chaincodes and both applications.

Reply:
Thanks for your contribution in fixing the issue that you saw !

actually, looking at ledger-query tutorial , `createAsset` here: https://github.com/hyperledger/fabric-samples/blob/df727853c84edec74dbbc2e0be37d93d9647d152/asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go#L104 
 
I would go for above order of args, in all CreateAsset:
- ID, color, owner : all three string
- size, appraisedValue : both int

my suggestion is to keep the order consistent by changing the mismatch in two places, at chaincode-javascript & chaincode-typescript `createAsset`.

If you close your 3 PR, & open a new PR to fix the bug in chaincode-javascript & chaincode-typescript (I will then pull in your changes after merge into my PR #226)

i.e. in `asset-transfer-basic/chaincode-javascript/lib/assetTransfer.js`  make it
```
 async createAsset(ctx, id, color, owner, size, appraisedValue)
```

same for typescript.

Reply:
Hi Sijo,  I see where you are coming from, but what you are suggesting is not quite as simple as changing as you suggest because all of the structs are configured per the way I reconciled to.  Even the struct in fabric-samples/asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go is: id, color, size, owner, appraisedValue.  My recommendation is to close your draft pr, then open a pr to fix any other mismatches so they are consistent across all chaincodes and applications.

Reply:
func (t *SimpleChaincode) CreateAsset(ctx contractapi.TransactionContextInterface, ID, color, owner string, size, appraisedValue int) error {

	exists, err := t.AssetExists(ctx, ID)
	if err != nil {
		return fmt.Errorf("Failed to get asset: " + err.Error())
	} else if exists {
		return fmt.Errorf("This asset already exists: " + ID)
	}

	objectType := "asset"
	asset := &asset{
		ObjectType:     objectType,
		ID:             ID,
		Color:          color,
		Size:           size,
		Owner:          owner,
		AppraisedValue: appraisedValue,
	}

Reply:
`func (t *SimpleChaincode) CreateAsset(ctx contractapi.TransactionContextInterface, ID, color, owner string, size, appraisedValue int) error {
	exists, err := t.AssetExists(ctx, ID)
	if err != nil {
		return fmt.Errorf("Failed to get asset: " + err.Error())
	} else if exists {
		return fmt.Errorf("This asset already exists: " + ID)
	}

	objectType := "asset"
	asset := &asset{
		ObjectType:     objectType,
		ID:             ID,
		Color:          color,
		Size:           size,
		Owner:          owner,
		AppraisedValue: appraisedValue,
	}`

Reply:
I don't quite understand. The sequence declared inside `asset := &asset` does not matter, but the sequence of func args does matter.

so if we do what I suggested, then the asset-transfer-basic will have same signature as https://github.com/hyperledger/fabric-samples/blob/df727853c84edec74dbbc2e0be37d93d9647d152/asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go#L104

Reply:
Technically it does not matter, but as these samples are going to be used in the new tutorials, consistency across the application and chaincode does matter.

Reply:
Let's standardize on:
id string, color string, size int, owner string, appraisedValue int

This is the order that Chris recommended, and I like it because it puts the immutable properties first, but we do need to fix the data types here.